viv_utils/emulator_drivers.py

```diff
@@ -108,6 +108,13 @@ def remove_hook(self, hook):
     def isCall(self, op):
         return bool(op.iflags & v_envi.IF_CALL)
 
+    def isJmpCall(self, op):
```

Comment:
does this mean "indirect jump" like `jmp [0x401000]`?

Reply:
yeah

Reply:
ah, but this isn't totally unknown, like `jmp [eax]`, its an indirect jump through an offset. i was going to say "lets rename this to `isIndirectJump`" but i think we need something even more specific. any ideas?

Reply:
`isIndirectMemJump`?

Reply:
yeah, that makes more sense

Reply:
now I wonder about `i386PcRelOper`, `i386RegMemOper`, etc.

Reply:
i'll merge and rename within my branch

Reply:
maybe, `return op.mnem == "jmp" and op.opers[0].isDeref()`

Reply:
see implementation of "tail call" handling here: https://github.com/williballenthin/viv-utils/pull/31/commits/1c16000a978408697d4b0ae83fd7f51de9b5c701

basically, if we recognize the target of a jmp is a function/API, then treat it like it was a call. otherwise, just like a normal jmp. so it becomes ok to do `jmp [CreateFile]` or even `jmp CreateFile` (if the absolute address is known).

Reply:
🏆